### PR TITLE
project proposal: SSZ Query Language for Erigon

### PR DESCRIPTION
**Pull request by SoarinSkySagar**

### CODE REVIEW

**Comment by RazorClient** [comment]
proposal is pretty good and lgtm
few points regarding the query lang are still debatable,however these can be resolved in the near term and dont need addressing now

File: projects/Erigon-SSZ-Query-Language.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+## Roadmap
+
+### Phase 1: Setup endpoint and write grammar parser to compile queries (week 6 - week 10)
+
+- Register the new `POST /eth/v0/execution/{block_id}/query` route in Erigon's RPC layer
+- Implement the query grammar parser for anchors, paths, filters, summaries, and aliases as defined in the specification.
+- Perform input validation on dummy data (since EL blocks are not SSZ encoded right now): verify the anchor is valid, every queried leaf is a descendant of its anchor, and each filter is valid for the leaf it targets
+
+### Phase 2: Wire up the API to Execution Blocks (week 11 - week 14)
+- Ensure that [Pureth: Execution Blocks for Erigon](./) project is complete to a point where actual EL block data can be used
```
Comment (taxmeifyoucan) [inline projects/Erigon-SSZ-Query-Language.md]: This sounds like a hard dependency on Jack's 7807 implementation. It's not clear here how you plan to collaborate ad help each other, it's good you have compatible roadmap but it needs to be managed well not to become blocker in your own

File: projects/Erigon-SSZ-Query-Language.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+- End-to-end and cross-verification tests: independently recompute `hash_tree_root` from returned witnesses to confirm proof correctness.
+- Benchmark response size and server-side cost of multiproof vs. per-field proofs and full-object fetches, validating the efficiency claim.
+- Test edge cases: empty results, invalid anchors/paths/filters, syncing node (503), and unfinalized blocks.
+- Documentation of the query grammar, endpoint usage, and examples.
+
+## Possible challenges
+
+- Execution blocks are not implemented in Erigon right now, and this project depends on them. A delay in their implementation project may delay this project as well.
+- There is no formal EIP for SSZ-QL yet, so the specification may be unstable right now and could keep changing actively as requirements evolve.
+- A concrete method is needed for proving the correctness of filtered results.
```
Comment (taxmeifyoucan) [inline projects/Erigon-SSZ-Query-Language.md:163]: Any proposals for this? Otherwise it's trusted model

**Review by taxmeifyoucan — COMMENTED** [review]
Thanks Sagar! The project looks sound and the proposal is well written. My main issue is still the overlap with the other project for Grandine, roadmaps overlap and I am not sure if it's possible to deliver both in parallel, huge amount of work even for multiple people.

File: projects/Erigon-SSZ-Query-Language.md (hunk 2 above)
Comment (SoarinSkySagar) [reply projects/Erigon-SSZ-Query-Language.md:163]: Yes! I have proposed it here:
https://hackmd.io/@SoarinSkySagar/rkwwvAMEzg

I have linked this below in resources section
